runtime/codegen/registry.go

```diff
@@ -146,3 +147,35 @@ func ComponentConfigValidator(path, cfg string) error {
 	}
 	return nil
 }
+
+// CallEdge records that fact that the Caller component uses the
+// Callee component. Both types are types of the corresponding
+// component interfaces.
+type CallEdge struct {
+	Caller reflect.Type
+	Callee reflect.Type
+}
+
+// CallGraph returns the component call graph (as a list of CallEdge values).
+func CallGraph() []CallEdge {
+	var result []CallEdge
+	for _, reg := range Registered() {
+		impl := reflect.TypeOf(reg.New())
```

Comment:
Is creating a new impl instance always a safe/cheap thing to do? In my upcoming change, every weavelet will be calling this function on startup.

An alternative is to change the codegen code to return the impl reflect.Type directly.

Reply:
I was doing that in this change but it brought in a bunch of unrelated changes. How about I make the switch to that in another change?